File: cpp/217. Contains Duplicate.cpp
//check in unordered_set, if element exists, return true else add the element
//return false after the loop ends
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int n = nums.size();
        unordered_set<int> s;
        for(int i=0;i<n;i++){
            if(s.find(nums[i])!=s.end()){
                return true;
            }
            else
                s.insert(nums[i]);
        }
        return false;
    }
};

File: Python/217. Contains Duplicate.py
#check in unordered_set, if element exists, return true else add the element
#return false after the loop ends
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        hashset = set()
        for x in nums:
            if x in hashset:
                return True
            hashset.add(x)
        return False